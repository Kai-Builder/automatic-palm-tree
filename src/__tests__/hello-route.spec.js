import ComponentToTest from '../hello-route';
import Hello from '../hello';
import React from 'react';
import {
  Route
} from 'react-router-dom';
import {
  shallow
} from 'enzyme';

describe('HelloRoute', () => {

  const requiredProps = () => ({});
  const render = (props = requiredProps()) => shallow(<ComponentToTest {...props}/>);

  test('component renders', () => {

    const wrapper = render();

    expect(wrapper).toMatchSnapshot();

  });

  test('component to be open for non authorized users', () => {

    const wrapper = render();

    expect(wrapper.type()).toEqual(Route);
    expect(wrapper).toMatchSnapshot();

  });

  test('route renders Hello', () => {

    const wrapper = render();

    expect(wrapper.props().component).toEqual(Hello);
    expect(wrapper).toMatchSnapshot();

  });

  test('path is correct', () => {

    const wrapper = render();

    expect(wrapper.props().path).toEqual('/');
    expect(wrapper).toMatchSnapshot();

  });

});
