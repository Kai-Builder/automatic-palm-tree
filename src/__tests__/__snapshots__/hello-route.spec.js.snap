// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloRoute component renders 1`] = `
<Route
  component={[Function]}
  path="/"
/>
`;

exports[`HelloRoute component to be open for non authorized users 1`] = `
<Route
  component={[Function]}
  path="/"
/>
`;

exports[`HelloRoute path is correct 1`] = `
<Route
  component={[Function]}
  path="/"
/>
`;

exports[`HelloRoute route renders Hello 1`] = `
<Route
  component={[Function]}
  path="/"
/>
`;
